File: src/main/core/MapEntity.py
from typing import Dict

class MapEntity:
    def __init__(self, name: str, population: int, demographics: Dict[str, float]):
        self.name = name
        self.population = population
        self.demographics = demographics
    
    def __str__(self) -> str:
        return f"name: {self.name}, population: {self.population}, demographics: {self.demographics}"

File: src/main/core/gathering_data/MapEntity.cpp
#include "MapEntity.h"

MapEntity::MapEntity(std::string name, int population, std::map<std::string, float> demographics) {
    this->name = name;
    this->population = population;
    this->demographics = demographics;
}

std::string MapEntity::getName() const {
    return this->name;
}

int MapEntity::getPopulation() const {
    return this->population;
}

std::map<std::string, float> MapEntity::getDemographics() const {
    return this->demographics;
}